core/cont/src/TClassTable.cxx

```diff
@@ -343,17 +380,19 @@ namespace ROOT { class TForNamespace {}; } // Dummy class to give a typeid to na
 void TClassTable::Add(const char *cname, Version_t id,  const std::type_info &info,
                       DictFuncPtr_t dict, Int_t pragmabits)
 {
-   if (!gClassTable)
-      new TClassTable;
-
    if (!cname || *cname == 0)
       ::Fatal("TClassTable::Add()", "Failed to deduce type for '%s'", info.name());
 
+   std::lock_guard<std::mutex> lock(GetClassTableMutex());
+
+   if (!gClassTable)
+      new TClassTable;
+
```

Comment:
That is correct.  It is unlikely that it would be worth changing this as:
* some (user) code might be calling the static interface via the object.
* the value of `gClassTable` can be as a way to know if the table has been created
* the `TClassTable` constructor does a lot of initialization.